Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2014-02-13
Initial Package Version: 20130530
Upstream Status: (TeX Live) - unknown
Origin: Upstream asymptote [ 2.24 ]
Description: Fixes scope error with (at least) g++-4.8 :
interact.cc: In function ‘char* interact::readpipeline(const char*)’:
interact.cc:94:25: error: ‘strlen’ was not declared in this scope
   } while(buf[strlen(buf)-1] != '\n');
                         ^
make: *** [interact.o] Error 1


--- a/utils/asymptote/interact.cc	2014-02-12 17:22:20.598967280 +0000
+++ b/utils/asymptote/interact.cc	2014-02-12 17:22:34.042882164 +0000
@@ -85,20 +85,10 @@
 
 char *readpipeline(const char *prompt)
 {
-  const int max_size=256;
-  static char buf[max_size];
-  ostringstream s;
-  do {
-    if(fgets(buf,max_size-1,fin) == NULL) break;
-    s << buf;
-  } while(buf[strlen(buf)-1] != '\n');
-  return StrdupMalloc(s.str());
-  /* Simpler version (requires POSIX 2008; temporarily removed for TeXLive 2013):
   char *line=NULL;
   size_t n;
-  n=getline(&line,&n,fin);
+  getline(&line,&n,fin);
   return line;
-  */
 }
   
 void pre_readline()
